test/testutils.jl

```diff
@@ -0,0 +1,170 @@
+
+# Return the "paths" of objects that exhibit differences between the two json inputs.
+# By default, we clean the results from some optional projjson objects and minor
+# discrepancies with GDAL's output. Set `exact = true` to disable that behavior.
+# Note: diffpaths uses isapprox to compare numbers to avoid false negatives
+function deltaprojjson(json1, json2; exact=false)
+  diffpaths = finddiffpaths(json1, json2)
+
+  # return full diff in case of exact comparison
+  exact && return diffpaths
+
+  # paths to ignore in approximate comparison
+  # bbox, area, scope, ... are not required to
+  # fully describe the coordinate reference system
+  pathstoignore = ["bbox", "area", "scope", "usages", "\$schema"]
+
+  # occasionally GDAL's ellipsoid doesn't have inverse_flattening and instead has semi_minor_axis
+  # that is calculated from semi_major_axis * (1 - 1/inverse_flattening)
+  push!(pathstoignore, "datum.ellipsoid.semi_minor_axis")
+  push!(pathstoignore, "datum.ellipsoid.inverse_flattening")
+
+  # we don't have the optional node base_crs.coordinate_system in our WKT, in contrast with GDAL
+  push!(pathstoignore, "base_crs.coordinate_system")
+
+  # delete insignificant json objects that are problematic for comparison testing
+  for p in pathstoignore
+    index = findfirst(endswith(p), diffpaths)
+    if !isnothing(index)
+      deleteat!(diffpaths, index)
+    end
+  end
+
+  return diffpaths
+end
+
+# Find differences between two dictionaries and return the paths to those differences as json dot-notation strings.
+# This function recursively compares two dictionaries and returns a vector of string paths pointing to the differences found.
+# For numerical values, uses `isapprox` for comparison to avoid false negatives.
+#
+# Example:
+#
+#    julia> d1 = Dict(:A=>[0,20], :B=>3, :C=>4)
+#    julia> d2 =  Dict(:A=>[10,20], :C=>4)
+#    julia> finddiffpaths(d1, d2)
+#    2-element Vector{String}:
+#     ".A[1]"
+#     ".B"
+function finddiffpaths(d1::Dict, d2::Dict, path="")
+  paths = String[]
+  allkeys = union(keys(d1), keys(d2))
+  for key in allkeys
+    newpath = string(path, ".", key)
+    if haskey(d1, key) && haskey(d2, key)
+      if !isequalfield(d1[key], d2[key])
+        append!(paths, finddiffpaths(d1[key], d2[key], newpath))
+      end
+    else
+      push!(paths, newpath)
+    end
+  end
+
+  return paths
+end
+
+function finddiffpaths(v1::Vector, v2::Vector, path)
+  paths = String[]
+  minlen = min(length(v1), length(v2))
+  maxlen = max(length(v1), length(v2))
+
+  for i in 1:minlen
+    if !isequalfield(v1[i], v2[i])
+      append!(paths, finddiffpaths(v1[i], v2[i], "$(path)[$(i)]"))
+    end
+  end
+  # extra elements, doesn't matter which vector
+  for i in (minlen + 1):maxlen
+    push!(paths, "$(path)[$(i)]")
+  end
+  return paths
+end
+
+function finddiffpaths(v1, v2, path)
+  isequalfield(v1, v2) ? nothing : [path]
+end
+
+# helper function to compare fields in JSON objects
```

Comment:
Could you please share an example where this comparison needs to be approximate? I am worried that we might be hitting a bug because these floating point values should match to the last digits. We copy/paste all digits from the EPSG database, and they usually matter.

Reply:
When you think you've seen every possible weird thing about GIS, this shows up 🤦🏽‍♂️ 

We should add a special test just for 3407, with a clear comment pointing to the two WKT sources. I understand GDAL is using some old WKT version.

Reply:
This is doesn't have to do with 3407 specifically. if you use `isequal` instead of `isapprox` 5 codes fail out 53 (testset shows 106 tests because each code is tested twice). And it seems that the 10% ratio holds for the whole of the dataset.
The testing with `isapprox` was an important part of development.
Note: the floating point discrepancies are never a result of us calculating something, it is always WKT vs WKT. I've tracked down multitude of them during development.  In fact there is NO floating point calculations done in the wkt2json code at all. I once had a `round` call but it has since been removed.

Reply:
Got it. If these are all noise in WKT vs. WKT we should be fine with the use of `isapprox` 👍🏽